src/meds_torch/data/components/multiwindow_pytorch_dataset.py

```diff
@@ -14,6 +14,131 @@
+    ...     windows_data,
+    ...     windows_to_fuse=["pre", "post"],
+    ...     fused_window_name="fused"
+    ... )
+    >>> # Check lengths tracking
+    >>> fused["LENGTHS//static_values"]  # Two windows with lengths 2 and 2
+    [2, 2]
+    >>> fused["LENGTHS//embeddings"]  # Two windows with sequence lengths 3 and 2
+    [3, 2]
+    >>> fused["LENGTHS//codes"]  # List lengths: 3 and 2
+    [3, 2]
+    >>> fused["LENGTHS//scalar"]  # Scalar values: count as length 1 each
+    [1]
+    >>> # Check concatenated values
+    >>> torch.equal(fused["static_values"],
+    ...            torch.cat([torch.tensor([1.0, 2.0]),
+    ...                      torch.tensor([3.0, 4.0])]))
+    True
+    >>> # Check 2D tensor concatenation along sequence dimension
+    >>> torch.equal(fused["embeddings"],
+    ...            torch.cat([torch.ones(2, 3, 2),
+    ...                      torch.ones(2, 2, 2) * 2], dim=1))
+    True
+    >>> fused["codes"]  # Lists are concatenated
+    ['A', 'B', 'C', 'D', 'E']
+    >>> fused["scalar"]  # First scalar value is kept
+    42
+    >>> # Test error handling
+    >>> fuse_window_data(windows_data, ["nonexistent"], "fused")  # doctest: +IGNORE_EXCEPTION_DETAIL
+    Traceback (most recent call last):
+    ValueError: Window nonexistent specified in windows_to_fuse not found in data
+    """
+    if not fused_window_name:
+        raise ValueError("fused_window_name must not be empty")
+
+    fused_window = {}
+    lengths_tracking = {}
+
+    # Process each window in the specified order
+    for window in windows_to_fuse:
+        if window not in windows_data:
+            raise ValueError(f"Window {window} specified in windows_to_fuse not found in data")
+
+        window_data = windows_data[window]
+
+        # Process each key in the window data
+        for key, data in window_data.items():
+            # Initialize containers in fused window
+            if key not in fused_window:
+                lengths_tracking[f"LENGTHS//{key}"] = []
+
+            if isinstance(data, torch.Tensor):
+                # For tensors, concatenate along the appropriate dimension
+                concat_dim = int(len(data.shape) > 1)
+                lengths_tracking[f"LENGTHS//{key}"].append(data.shape[concat_dim])
+                if key not in fused_window:
+                    fused_window[key] = data
+                else:  # 2D or higher tensor
+                    if key not in fused_window:
+                        fused_window[key] = data
+                    else:
+                        fused_window[key] = torch.cat([fused_window[key], data], dim=concat_dim)
+
+            elif isinstance(data, list):
+                # For lists, extend the fused window list
+                fused_window.setdefault(key, []).extend(data)
+                lengths_tracking[f"LENGTHS//{key}"].append(len(data))
+
+            elif isinstance(data, (int, float, datetime.datetime)):
+                # For scalars, use the first value
```

Comment:
This doesn't seem super robust... what is this case for? Why is it allowed?

Reply:
This is mainly for propogating prediction_time and subject_id